### ISSUE TITLE
Accessibility: app settings dialog buttons not accessible

### ISSUE THREAD

**Issue by shortwavesurfer2009** [glitch-soc/mastodon#1943]
### Steps to reproduce the problem

1. Open an instance running glitch social
2. Enable a screen reader (android talkback gor me in Settings > Accessibility)
3. Open the app preferences box from the sidebar
4. Tap the icons in the left row

### Expected behaviour

the buttons should read as "general", "compose", "collapsed toots", "media", "mastodon settings", and "close app preferrences"

### Actual behaviour

buttons are not labeled

### Detailed description

Most of the links on the left side of this box
![Screenshot_20221116-024305_Fennec_1](https://user-images.githubusercontent.com/116814522/202138247-e2dbad2d-62f6-4d84-9bb8-e83e49af5760.png)
read as "button" when using a screen reader such as talkback. The only two exceptions are "cw" and "link" on the gear icon for standard mastodon settings

### Specifications

Firefox 105
Android 11
Mastodon v4.0.1+glitch

**Comment by ClearlyClaire** [comment]
I have added labels/titles to those buttons in #1949. I have not marked this issue as closed though because my understanding is that a significant amount of work is still required to make that dialog really accessible.

I think the whole modal could make use of tabpanel/tab/tablist roles, but that isn't completely straightforward to do correctly.
